crates/ruff_cli/src/commands/add_noqa.rs

```diff
@@ -56,7 +57,12 @@ pub(crate) fn add_noqa(
                 .and_then(|parent| package_roots.get(parent))
                 .and_then(|package| *package);
             let settings = resolver.resolve(path, pyproject_config);
-            match add_noqa_to_path(path, package, source_type, settings) {
+            let Ok(LintSource(source_kind)) = LintSource::try_from_path(path, source_type) else {
+                // TODO(dhruvmanila): Display the error to the user.
+                error!("Failed to extract source from {}", path.display());
+                return None;
+            };
+            match add_noqa_to_path(path, package, &source_kind, source_type, settings) {
```

Comment:
Maybe unify the error handling either by using `LintSource::try_from_path(...).and_then(|source| add_noqa_to_path(...))` or extracing into its own function and calling it.